src/metabase/query_processor/util/add_alias_info.clj

```diff
@@ -0,0 +1,198 @@
+(ns metabase.query-processor.util.add-alias-info
+  (:require [clojure.walk :as walk]
+            [metabase.mbql.schema :as mbql.s]
+            [metabase.mbql.util :as mbql.u]
+            [metabase.query-processor.error-type :as qp.error-type]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.util.i18n :refer [tru]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]))
+
+(defn- remove-namespaced-options [options]
+  (not-empty (into {}
+                   (remove (fn [[k _]]
+                             (when (keyword? k)
+                               (namespace k))))
+                   options)))
+
+(s/defn normalize-clause :- mbql.s/FieldOrAggregationReference
+  "Normalize a `:field`/`:expression`/`:aggregation` clause by removing extra info so it can serve as a key for
+  `:qp/refs`."
+  [clause :- mbql.s/FieldOrAggregationReference]
+  (mbql.u/match-one clause
+    :field
+    (mbql.u/update-field-options &match (comp remove-namespaced-options
+                                              #_remove-default-temporal-unit
```

Comment:
is this intended to be commented out?

Reply:
Yeah. I removed that (will push changes shortly)